honeybee_energy/simulation/daylightsavings.py

```diff
@@ -0,0 +1,191 @@
+# coding=utf-8
+"""EnergyPlus Daylight Saving Time Period."""
+from __future__ import division
+
+from ..reader import parse_idf_string
+from ..writer import generate_idf_string
+
+from ladybug.analysisperiod import AnalysisPeriod
+from ladybug.dt import Date
+
+
+class DaylightSavingTime(object):
+    """EnergyPlus Daylight Saving Time Period.
+
+    Properties:
+        * start_date
+        * end_date
+        * is_leap_year
+    """
+    __slots__ = ('_start_date', '_end_date')
+
+    def __init__(self, start_date=Date(3, 12), end_date=Date(11, 5)):
+        """Initialize DaylightSavingTime.
+
+        Args:
+            start_date: A ladybug Date object for the start of daylight saving time.
+                Must be before the end date and have a leap_year property matching the
+                end_date. Default: 12 Mar (daylight savings in the US in 2017)
+            end_date: A ladybug Date object for the end of daylight saving time.
+                Must be after the start date and have a leap_year property matching the
+                start_date. Default: 5 Nov (daylight savings in the US in 2017)
+        """
+        # process the dates
+        if start_date is not None:
+            self._check_date(start_date, 'start_date')
+            self._start_date = start_date
+        else:
+            self._start_date = Date(3, 12)
+        self.end_date = end_date
+
+    @property
+    def start_date(self):
+        """Get or set a ladybug Date object for the start of the period."""
+        return self._start_date
+
+    @start_date.setter
+    def start_date(self, value):
+        if value is not None:
+            self._check_date(value, 'start_date')
+            self._start_date = value
+        else:
+            self._start_date = Date(1, 1)
+        self._check_start_before_end()
+
+    @property
+    def end_date(self):
+        """Get or set a ladybug Date object for the end of the period."""
+        return self._end_date
+
+    @end_date.setter
+    def end_date(self, value):
+        if value is not None:
+            self._check_date(value, 'start_date')
+            self._end_date = value
+        else:
+            self._end_date = Date(12, 31)
+        self._check_start_before_end()
+
+    @property
+    def is_leap_year(self):
```

Comment:
Ok, that's a fair suggestion since you are right that I don't really anticipate people using the `DaylightSavings` object outside of the `Runperiod` object. Let me see if I can change it.

Reply:
Ok, I removed this property here and the `RunPeriod` object now handles all coordination of leap year dates.